search/api.py

```diff
@@ -300,4 +307,14 @@ def document_needs_updating(enrollment):
         return True
     serialized_enrollment = serialize_program_enrolled_user(enrollment)
     del serialized_enrollment['_id']
-    return serialized_enrollment != document['_source']
+    source = document['_source']
+
+    if serialized_enrollment != source:
```

Comment:
Should we have this behind a flag so we can only turn it on when we need to debug? I'm not sure there's utility in logging every diff we see.